AndroidXCI/lib/src/main/kotlin/dev/androidx/ci/testRunner/TestRunnerServiceImpl.kt

```diff
@@ -206,13 +207,57 @@ internal class TestRunnerServiceImpl internal constructor(
         }
     }
 
+    private suspend fun findScreenshotFiles(
+        resultPath: GcsPath,
+        testIdentifiers: List<TestRunnerService.TestIdentifier>
+    ): Map< TestRunnerService.TestIdentifier, List<TestRunnerService.TestCaseArtifact>> {
+        val screenshotArtifactsBlobs = mutableMapOf<TestRunnerService.TestIdentifier, MutableList<TestRunnerService.TestCaseArtifact>>()
+        val screenshotArtifacts: Map<TestRunnerService.TestIdentifier, List<TestRunnerService.TestCaseArtifact>>
+        val testNames = testIdentifiers.associateBy { testIdentifier ->
+            "${testIdentifier.className}_${testIdentifier.name}"
+        }
+        val testRunNumber = testIdentifiers.first().runNumber
+        fun BlobVisitor.fullDeviceId() = relativePath.substringBefore('/', "")
+        googleCloudApi.walkEntires(
+            gcsPath = resultPath
+        ).forEach { visitor ->
+            val runNumber = DeviceRun.create(visitor.fullDeviceId()).runNumber
+            if (runNumber == testRunNumber) {
+                val testName = testNames.keys.find { testName ->
+                    visitor.fileName.startsWith(testName)
+                }
+                val testIdentifier = testNames[testName]
+                if (testIdentifier != null) {
+                    screenshotArtifactsBlobs.getOrPut(testIdentifier) {
+                        mutableListOf()
+                    }.add(
+                        TestRunnerService.TestCaseArtifact(
+                            ResultFileResourceImpl(visitor),
+                            visitor.fileName.substringAfterLast(".")
```

Comment:
so if we will just use the extension for type, I think you can ignore my `ResourceType` suggestion above but still maybe declare known constants as strings?

alternativelys, we can add a function to ResourceType to create an insteance from a string. up to you.

Reply:
Added companion object ResourceType to TestArtifact class with const strings.